packages/api-client/tests/test-granules.js

```diff
@@ -99,6 +97,42 @@ test.serial('waitForGranules retries on status codes other than 500, 200, then t
 });
 
 
+test.serial('waitForGranules retries if status does not match provided status', async (t) => {
+  let retryCount = 0;
+  const retries = 2;
+  const getGranuleRevert = granulesRewire.__set__('getGranule', async ({ callback }) => {
+    retryCount += 1;
+    return callback(retryCount);
+  });
+  let revertCallback;
+  try {
+    const callback = (retry) => {
+      if (retry < 3) {
```

Comment:
Maybe this doesn't make sense, but should we do `retry < retries + 1` here? Otherwise there's no connection between the number of retries you're specifying for the test below and the mocking?

Reply:
It makes sense, good catch : 8cf91d6c5